<a href="https://colab.research.google.com/github/tomlinsw9207/DATA-110/blob/main/Cannon_Elliott_Lab2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Lab 2: Starbucks

**Be sure to save a copy of this notebook to your drive before you start working!!** You should also save a copy of the data set Starbucks_DrinkMenu_Expanded.csv to your own drive by right-clicking on the file and choosing "Make a copy."

Please complete this assignment by answering questions and writing code as indicated in each section of this notebook. When you are done,  
1.   save your work to a **PDF** for submission on **Canvas** (remember that you can do this by printing to a PDF), and
2.   download the **.ipynb** file of your Colab notebook for submission on **Canvas**.

Please name your file "Lastname_firstname_Lab2.pdf" and ''Lastname_firstname_Lab2.ipynb''

**Name:** Elliot Cannon

**Section:** DATA 110-001


In [ ]:
# import pandas an mount your google drive

import pandas as pd

from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# Part 1: Load the data and review the overview

Q1: Write code to display the information about each column including its data type and prints the first five rows of the dataframe.


In [ ]:
# load the data into a pandas dataframe (remember to change data_path)

data_path = '/content/drive/My Drive/UNC/26-27/DATA 110/Lab 2/Starbucks_DrinkMenu_Expanded.csv'
Starbucks = pd.read_csv(data_path)

# Write your code to preview data
Starbucks
Starbucks.info()
Starbucks.shape
Starbucks.dtypes




<class 'pandas.core.frame.DataFrame'>
RangeIndex: 242 entries, 0 to 241
Data columns (total 19 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Beverage_category       242 non-null    object 
 1   Beverage                242 non-null    object 
 2   Size                    242 non-null    object 
 3   Milk                    215 non-null    object 
 4   Calories                242 non-null    int64  
 5   Total_Fat_g_            242 non-null    float64
 6   Trans_Fat_g_            242 non-null    float64
 7   Saturated_Fat_g_        242 non-null    float64
 8   Sodium_mg_              242 non-null    int64  
 9   Total_Carbohydrates_g_  242 non-null    int64  
 10  Cholesterol_mg_         242 non-null    int64  
 11  Dietary_Fibre_g_        242 non-null    int64  
 12  Sugars_g_               242 non-null    int64  
 13  Protein_g_              242 non-null    float64
 14  Vitamin_A_DV_           242 non-null    fl

,0
Beverage_category,object
Beverage,object
Size,object
Milk,object
Calories,int64
Total_Fat_g_,float64
Trans_Fat_g_,float64
Saturated_Fat_g_,float64
Sodium_mg_,int64
Total_Carbohydrates_g_,int64


Use the code and output above to answer the following questions:

Q2: how many columns and rows are in this data set?

A: 242 rows and 19 columns

Q3: Looking at the types of the columns in the dataframe, what do you notice about the type for Caffeine_mg_?  What were you expecting the data type to be?

A: The type is an object, but I would expect it to be an integer type.

Q4: Write code to convert Caffeine_mg_ to the appropriate data type using ```pd.to_numeric()```. (You can use Google to learn how to use the function ```pd.to_numeric()```)


In [ ]:
# Write code to convert Caffeine_mg_ to the appropriate data type

Starbucks['Caffeine_mg_'] = pd.to_numeric(Starbucks['Caffeine_mg_'], errors='coerce', downcast='integer')

Starbucks.dtypes
Starbucks.info()




<class 'pandas.core.frame.DataFrame'>
RangeIndex: 242 entries, 0 to 241
Data columns (total 19 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Beverage_category       242 non-null    object 
 1   Beverage                242 non-null    object 
 2   Size                    242 non-null    object 
 3   Milk                    215 non-null    object 
 4   Calories                242 non-null    int64  
 5   Total_Fat_g_            242 non-null    float64
 6   Trans_Fat_g_            242 non-null    float64
 7   Saturated_Fat_g_        242 non-null    float64
 8   Sodium_mg_              242 non-null    int64  
 9   Total_Carbohydrates_g_  242 non-null    int64  
 10  Cholesterol_mg_         242 non-null    int64  
 11  Dietary_Fibre_g_        242 non-null    int64  
 12  Sugars_g_               242 non-null    int64  
 13  Protein_g_              242 non-null    float64
 14  Vitamin_A_DV_           242 non-null    fl

# Part 2: Handling missing values

Q5: Do any of the columns in this data set appear to have missing values?  If so, which ones?

A: Yes, there are values missing from the Caffeine_mg_ and Milk columns

Q6: Write code to calculate the percentage of missing values for each column. (Hint:  Use ```isna()```)

Q7: Which column has the highest percentage of missing values? What percent are missing?

A : Milk has the highest percentage missing. It's 11.157% missing

In [ ]:
# calculate the percentage of missing values for each column.
Starbucks.isna()
Starbucks.isna().sum()
Starbucks.isna().sum()/len(Starbucks)

,0
Beverage_category,0.000000
Beverage,0.000000
Size,0.000000
Milk,0.111570
Calories,0.000000
Total_Fat_g_,0.000000
Trans_Fat_g_,0.000000
Saturated_Fat_g_,0.000000
Sodium_mg_,0.000000
Total_Carbohydrates_g_,0.000000


Q8: Do you think it would be appropriate to impute the missing values for Milk with the most frequently occurring value in the column?  Why or why not?

A: No, because the data represents the make up of a beverage, so filling it in with what is frequently occuring would be very innacturate. The recipes might also call for no milk, so putting a frequently occuring value of milk for a missing value could make the data misrepresentative.

Q9: Write code in the cell below to replace the missing values for milk with "None" and then list the unique values of milk to confirm that your code was successful

In [ ]:
# Use fillna to impute missing values

Starbucks['Milk'].fillna('None', inplace=True)
Starbucks['Milk'].unique()

# Check for missing values again (Hint - value_counts())
Starbucks['Milk'].value_counts()

/tmp/ipykernel_2493/4080192110.py:3: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  Starbucks['Milk'].fillna('None', inplace=True)


,count
Milk,
Nonfat Milk,83
Soymilk,66
2% Milk,50
None,27
Whole Milk,16


# Part 3: Analyzing grande drinks

Our goal is to examine the nutritional content of Starbucks drinks.  In order to control for the size of the drink (since larger drinks will obviously have more sugar, calories, etc than smaller drinks), we will restrict our analysis to one drink size.  Research has shown that "Grande" is the most popular drink size at Starbucks.  

Q10: Write code to create a new dataframe that contains only drinks where Size = Grande and print the firt 5 rows (head()) below.



In [ ]:
# Create a Grande dataframe
Size = Starbucks[Starbucks['Size'] == 'Grande']
Size.head()

,Beverage_category,Beverage,Size,Milk,Calories,Total_Fat_g_,Trans_Fat_g_,Saturated_Fat_g_,Sodium_mg_,Total_Carbohydrates_g_,Cholesterol_mg_,Dietary_Fibre_g_,Sugars_g_,Protein_g_,Vitamin_A_DV_,Vitamin_C_DV_,Calcium_DV_,Iron_DV_,Caffeine_mg_
2,Coffee,Brewed Coffee,Grande,None,5,0.1,0.0,0.0,0,10,0,0,0,1.0,0.00,0.00,0.00,0.00,330.0
10,Classic Espresso Drinks,Caffe Latte,Grande,Nonfat Milk,130,0.3,0.2,0.0,5,150,19,0,18,13.0,0.20,0.00,0.40,0.00,150.0
11,Classic Espresso Drinks,Caffe Latte,Grande,2% Milk,190,7.0,3.5,0.2,30,170,19,0,17,12.0,0.20,0.02,0.40,0.00,150.0
12,Classic Espresso Drinks,Caffe Latte,Grande,Soymilk,150,5.0,0.5,0.0,0,130,13,1,8,10.0,0.15,0.00,0.40,0.15,150.0
22,Classic Espresso Drinks,Caffe Mocha (Without Whipped Cream),Grande,Nonfat Milk,220,2.5,1.5,0.0,5,125,43,2,34,13.0,0.20,0.00,0.35,0.25,175.0


Q11: What is the average calorie count of the Grande drinks?

A: 216

In [ ]:
# find the average calorie count

Grande = Size['Calories']
Grande.mean()

np.float64(216.3013698630137)

Q12: Which drink has the highest sugar content?  The highest caffeine content? You will need to write code to find the drink with the highest sugar content and caffeine content.

A: 65 highest sugar and 330 highest caffein

In [ ]:
# Which drink has the highest sugar content?

Sugar = Size['Sugars_g_']
Sugar.max()
print(Sugar.max())

# Which drink has the highest cafeine content?

Caffeine = Size['Caffeine_mg_']
Caffeine.max()
print(Caffeine.max())

65
330.0


Q13: Generate the average calorie count for each beverage category.  If you are craving Starbucks but trying to be calorie conscious, which category of beverages should you choose from? What would be the worst choice in this situation? (Hint:  ```groupby()```)

A: best choice would be coffee, worst choice is smoothies.

In [ ]:
# generage average calorie count by beverage category
Calorie_Count = Size.groupby('Beverage_category')['Calories'].mean()
Calorie_Count


,Calories
Beverage_category,
Classic Espresso Drinks,162.500000
Coffee,5.000000
Frappuccino Blended Coffee,272.500000
Frappuccino Blended Creme,242.500000
Frappuccino Light Blended Coffee,160.000000
Shaken Iced Beverages,106.666667
Signature Espresso Drinks,281.000000
Smoothies,282.222222
Tazo Tea Drinks,203.076923
